frontend/packages/ceph-storage-plugin/src/components/namespace-store/namespace-store-form.tsx

```diff
@@ -0,0 +1,205 @@
+    providerDataReducer,
+    initialState,
+  );
+
+  const handleNsNameTextInputChange = (strVal: string) => {
+    if (strVal.length <= 43) {
+      setNsName(strVal);
+    }
+  };
+
+  const {
+    cancel,
+    className,
+    inProgress,
+    close,
+    errorMessage,
+    handlePromise,
+    redirectHandler,
+    namespace,
+  } = props;
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    /** Create a secret if secret ==='' */
+    let { secretName } = providerDataState;
+    const promises = [];
+    if (!secretName) {
+      secretName = nsName.concat('-secret');
+      const { secretKey, accessKey } = providerDataState;
+      const secretPayload = secretPayloadCreator(
+        provider,
+        namespace,
+        secretName,
+        accessKey,
+        secretKey,
+      );
+      providerDataDispatch({ type: 'setSecretName', value: secretName });
+      promises.push(k8sCreate(SecretModel, secretPayload));
+    }
+    /** Payload for ns */
+    const nsPayload: Payload = {
+      apiVersion: apiVersionForModel(NooBaaNamespaceStoreModel),
+      kind: NooBaaNamespaceStoreModel.kind,
+      metadata: {
+        namespace,
+        name: nsName,
+      },
+      spec: {
+        type: NOOBAA_TYPE_MAP[provider],
+        ssl: false,
+      },
+    };
+    if (externalProviders.includes(provider)) {
+      nsPayload.spec = {
+        ...nsPayload.spec,
+        [PROVIDERS_NOOBAA_MAP[provider]]: {
+          [BUCKET_LABEL_NOOBAA_MAP[provider]]: providerDataState.target,
+          secret: {
+            name: secretName,
+            namespace,
+          },
+        },
+      };
+    }
+    if (provider === BC_PROVIDERS.S3) {
+      // eslint-disable-next-line
+      nsPayload.spec['s3Compatible'] = {
+        // eslint-disable-next-line
+        ...nsPayload.spec['s3Compatible'],
+        endpoint: providerDataState.endpoint,
+      };
+    } else if (provider === BC_PROVIDERS.IBM) {
+      nsPayload.spec.ibmCos = { ...nsPayload.spec.ibmCos, endpoint: providerDataState.endpoint };
+    }
+    // Add region in the end
+    if (provider === BC_PROVIDERS.AWS) {
+      nsPayload.spec.awsS3 = { ...nsPayload.spec.awsS3, region: providerDataState.region };
+    }
+
+    promises.push(k8sCreate(NooBaaNamespaceStoreModel, nsPayload));
+    return handlePromise(Promise.all(promises), (resource) => {
+      redirectHandler ? redirectHandler(resource) : close();
+    });
+  };
+
+  return (
+    <Form
+      className={classNames('nb-endpoints-form', className)}
+      onSubmit={onSubmit}
+      noValidate={false}
+    >
+      <FormGroup
+        label={t('ceph-storage-plugin~Namespace Store Name')}
+        fieldId="namespacestore-name"
+        className="nb-endpoints-form-entry"
+        helperText={t(
+          'ceph-storage-plugin~A unique name for the namespace store within the project',
+        )}
+        isRequired
+      >
+        <Tooltip
+          content="Name can contain a max of 43 characters"
```

Comment:
externalize for i18n
```suggestion
          content="Name can contain a max of 43 characters"
```

Reply:
done